File: Python/visualization/code.py
# Visualization Libraries Demo
# Demonstrates: matplotlib, seaborn — different plot types

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# ============ CHANGE THESE ============
FILENAME = "data.csv"
NUM_COL1 = "col1"    # a numeric column
NUM_COL2 = "col2"    # another numeric column
CAT_COL = "category" # a categorical column
# ======================================

df = pd.read_csv(FILENAME)
print("Shape:", df.shape)
print(df.head())

# --- 1. Line Plot ---
plt.figure(figsize=(8, 4))
plt.plot(df[NUM_COL1], label=NUM_COL1)
plt.title("Line Plot")
plt.xlabel("Index")
plt.ylabel(NUM_COL1)
plt.legend()
plt.tight_layout()
plt.show()

# --- 2. Scatter Plot ---
plt.figure(figsize=(8, 4))
plt.scatter(df[NUM_COL1], df[NUM_COL2], alpha=0.6)
plt.title("Scatter Plot")
plt.xlabel(NUM_COL1)
plt.ylabel(NUM_COL2)
plt.tight_layout()
plt.show()

# --- 3. Histogram ---
plt.figure(figsize=(8, 4))
plt.hist(df[NUM_COL1], bins=20, edgecolor='black')
plt.title("Histogram")
plt.xlabel(NUM_COL1)
plt.ylabel("Frequency")
plt.tight_layout()
plt.show()

# --- 4. Bar Plot (Seaborn) ---
plt.figure(figsize=(8, 4))
sns.countplot(data=df, x=CAT_COL)
plt.title("Bar Plot - Category Counts")
plt.tight_layout()
plt.show()

# --- 5. Box Plot (Seaborn) ---
plt.figure(figsize=(8, 4))
sns.boxplot(data=df, x=CAT_COL, y=NUM_COL1)
plt.title("Box Plot")
plt.tight_layout()
plt.show()

# --- 6. Heatmap - Correlation ---
plt.figure(figsize=(8, 6))
numeric_df = df.select_dtypes(include='number')
sns.heatmap(numeric_df.corr(), annot=True, cmap='coolwarm', fmt='.2f')
plt.title("Correlation Heatmap")
plt.tight_layout()
plt.show()

# --- 7. Pair Plot ---
sns.pairplot(df.select_dtypes(include='number').iloc[:, :4])
plt.suptitle("Pair Plot", y=1.02)
plt.show()

print("All 7 plots generated successfully!")

"""
# --- CLEAN & PERFECT VISUALIZATION ---
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv("data.csv")

# 1. Scatter (Numeric vs Numeric)
sns.scatterplot(x=df["col1"], y=df["col2"])
plt.show()

# 2. Bar (Category Counts)
sns.countplot(x=df["category"], data=df)
plt.show()

# 3. Correlation Heatmap
sns.heatmap(df.select_dtypes(include='number').corr(), annot=True)
plt.show()
"""

File: R/06_Visualization/code.R
# Visualization Libraries in R using ggplot2

# ============ CHANGE THESE ============
FILENAME <- "data.csv"
NUM_COL1 <- "col1"     # numeric column
NUM_COL2 <- "col2"     # another numeric column
CAT_COL <- "category"  # categorical column
# ======================================

library(ggplot2)

df <- read.csv(FILENAME)
cat("Shape:", nrow(df), "x", ncol(df), "\n")
print(head(df))

# --- 1. Scatter Plot ---
print(ggplot(df, aes_string(x = NUM_COL1, y = NUM_COL2)) +
        geom_point(alpha = 0.6, color = "steelblue") +
        ggtitle("Scatter Plot") + theme_minimal())

# --- 2. Histogram ---
print(ggplot(df, aes_string(x = NUM_COL1)) +
        geom_histogram(bins = 20, fill = "steelblue", color = "black") +
        ggtitle("Histogram") + theme_minimal())

# --- 3. Bar Plot ---
print(ggplot(df, aes_string(x = CAT_COL)) +
        geom_bar(fill = "steelblue") +
        ggtitle("Bar Plot") + theme_minimal())

# --- 4. Box Plot ---
print(ggplot(df, aes_string(x = CAT_COL, y = NUM_COL1)) +
        geom_boxplot(fill = "lightblue") +
        ggtitle("Box Plot") + theme_minimal())

# --- 5. Line Plot ---
print(ggplot(df, aes_string(x = seq_len(nrow(df)), y = NUM_COL1)) +
        geom_line(color = "steelblue") +
        xlab("Index") + ggtitle("Line Plot") + theme_minimal())

# --- 6. Density Plot ---
print(ggplot(df, aes_string(x = NUM_COL1)) +
        geom_density(fill = "steelblue", alpha = 0.5) +
        ggtitle("Density Plot") + theme_minimal())

# --- 7. Correlation Heatmap (base R) ---
numeric_df <- df[sapply(df, is.numeric)]
cor_matrix <- cor(numeric_df, use = "complete.obs")
cat("\nCorrelation Matrix:\n")
print(round(cor_matrix, 2))
heatmap(cor_matrix, main = "Correlation Heatmap", scale = "none")

cat("\nAll plots generated!\n")
